### PR TITLE
Vitest and Playwright tests

### PR DESCRIPTION
**Pull request by derekadombek**
# Pull Request

## Description
[Provide a brief description of the changes implemented in this pull request.]

## Related Issues
[Link any related issues or tasks from your project management system.]

## Additional Notes
[Add any additional context or notes that reviewers should know about.]

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.
- [ ] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

**Review by arinkulshi-skylight — DISMISSED** [review]
LGTM

File: .github/workflows/frontend-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18'
+      - name: Install dependencies
+        run: npm install @rollup/rollup-linux-x64-gnu --save-optional
+      - name: Run Jest tests
```
Comment (zdeveloper) [inline .github/workflows/frontend-tests.yaml:23]: we don't use jest, we use vitest, i know its just a label, but its inaccurate

File: .github/workflows/frontend-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    defaults:
+        run:
+          working-directory: OCR/frontend
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18'
```
Comment (zdeveloper) [inline .github/workflows/frontend-tests.yaml:20]: we use node 20

File: .github/workflows/frontend-tests.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
```
Comment (zdeveloper) [inline .github/workflows/frontend-tests.yaml:37]: same node v20.16.0

**Review by arinkulshi-skylight — APPROVED** [review]

File: .github/workflows/frontend-tests.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Install dependencies
+        run: npm install @rollup/rollup-linux-x64-gnu --save-optional
```
Comment (zdeveloper) [inline .github/workflows/frontend-tests.yaml:22]: we dont need any special install commands
just `npm install`

**Review by zdeveloper — CHANGES_REQUESTED** [review]

**Review by zdeveloper — CHANGES_REQUESTED** [review]

**Review by zdeveloper — APPROVED** [review]
